table/block_based/filter_policy.cc

```diff
@@ -203,9 +216,30 @@ class XXPH3FilterBitsBuilder : public BuiltinFilterBitsBuilder {
     return rv;
   }
 
-  // A deque avoids unnecessary copying of already-saved values
-  // and has near-minimal peak memory use.
-  std::deque<uint64_t> hash_entries_;
+  Status MaybeVerifyHashEntriesChecksum() {
```

Comment:
Ideally, the values would be verified as we are iterating over them in construction. That would be faster and leave fewer spaces for undetected memory/CPU corruption. But I understand it is difficult to inject into the modular Ribbon construction code. So I'm ok with just a comment (TODO?) pointing out the potential improvement, or make the improvement for Bloom only, where construction CPU efficiency is more critical.

Actually, `ResetAndFindSeedToSolve` is a template function that can use a custom iterator. If we have a custom iterator that tracks the xor checksum as it iterates, that might work for both Bloom and Ribbon.

Reply:
Added to TODO - I will include Bloom as follow-up work too since changing it now changes the error injection approach in my test now.